### PR TITLE
Added subschema validation

### PR DESCRIPTION
**Pull request by iantimmis**
Added functionality to determine whether a VideoLabelsSchema is a subset of another VideoLabelsSchema. Likewise with ImageLabelsSchema.

```py
from eta.core.video import VideoLabelsSchema

ont_schema = VideoLabelsSchema.from_json('ontology.json')
cov_schema = VideoLabelsSchema.from_json('coverage.json')

print(ont_schema.is_valid_subset_schema(cov_schema)) # Prints True
```

### CODE REVIEW

File: eta/core/image.py [hunk 1]
```diff
@@ -485,6 +486,32 @@ def is_valid_object(self, obj):
         except (ImageLabelsSchemaError, AttributeContainerSchemaError):
             return False
 
+    def is_valid_subset_schema(self, schema):
+        '''Validates that the provided schema is a subset of self
+        
+        Args:
+            schema: an ImageLabelsSchema
+        Returns:
```
Comment (brimoor) [inline eta/core/image.py]: one space between `Args:` and `Returns:` sections

File: eta/core/image.py [hunk 2]
```diff
@@ -485,6 +486,32 @@ def is_valid_object(self, obj):
...
+            True/False
+        '''
+
+        # Validate frame attributes
+        for attr in schema.attrs:
+            if not self.is_valid_image_attribute(attr):
```
Comment (brimoor) [inline eta/core/image.py]: This doesn't work. `attr` is an `AttributeSchema`, but `is_valid_image_attribute` expects an `Attribute`

File: eta/core/image.py [hunk 3]
```diff
@@ -485,6 +486,32 @@ def is_valid_object(self, obj):
...
+                return False
+
+        # Validate objects
+        for obj in schema.objects:
+            if not self.is_valid_object_label(obj):
+                return False
+
+            # Validate object attributes
+            for attr in schema.objecs[obj].schema
```
Comment (brimoor) [inline eta/core/image.py]: typo: `objecs` -> `objects`

File: eta/core/image.py (hunk 2 above)
Comment (brimoor) [reply eta/core/image.py]: The same comment applies throughout this PR. No functionality yet exists for, e.g., checking whether the `categories` of one `CategoricalAttributeSchema` are a subset of another. That will have to be written

Comment (iantimmis) [reply eta/core/image.py]: Ah, gotcha. Thanks for the review. I'll work it out.

**Comment by brimoor** [comment]
@iantimmis you can drop this PR. I implemented it in https://github.com/voxel51/eta/pull/422.

In case you're interested, here are the relevant commits:
https://github.com/voxel51/eta/pull/422/commits/9b22e792cb4180f25d4b359ce8d6ed9e5fd25d37
https://github.com/voxel51/eta/pull/422/commits/bccf396eebb04047a37bd30a1a4727bc2ffbd0ed
https://github.com/voxel51/eta/pull/422/commits/780177e38619b1ac290201cf1444b9d51fdec494
https://github.com/voxel51/eta/pull/422/commits/42654d692137bb1d94014a3abb34ecba7b5a312d

**Pull request closed without merging**